src/lib/mdns/minimal/core/QName.cpp

```diff
@@ -134,5 +134,22 @@ const uint8_t * SerializedQNameIterator::FindDataEnd()
     return nullptr;
 }
 
+bool SerializedQNameIterator::operator==(const FullQName & other) const
+{
+    SerializedQNameIterator self = *this; // allow iteration
+    size_t idx                   = 0;
+
+    while ((idx < other.nameCount) && self.Next())
+    {
+        if (strcmp(self.Value(), other.names[idx]) != 0)
```

Comment:
Should use `strncmp`. Also, I see this is all stored as nul-terminated strings. Is there a reason strings were not stored in mDNS-like format (length+value)?

Reply:
Reply parsing are pascal strings, for our code it is easier to have `const QNamePart names[] = { "foo", "bar", "baz"}` than something like:
```
const uint8_t names[] = { 3, 'f', 'o', 'o', 3, 'b', 'a', 'r', 3, 'b', 'a', 'z', 0};
```

We may at some point unify them. Would require a few passes though.
I would presonally prefer client not to care about encoding, but right now they have to and it is inconvenient.